Documentation

bhilbert4 opened the pull request:
This PR begins to add some basic coherent documentation in order to help the power users who are going to start playing with `jwst_reffiles` in the next week or so. There is still more that needs to be added, but hopefully I've covered the basics here.

bhilbert4 commented:
Assuming that Armin is going to be the one to address the questions in #45, and that we probably don't want to deal with having him to push his changes to my branch before merging, my thought is that we should merge this and then have Armin open a new PR when he has time to answer those questions. @arminrest does that make sense to you? If so I'll probably ask @aliciacanipe or @bsunnquist to take a quick look at these docs to look for spelling/grammar/glaring omissions, and then merge when they are happy.

bhilbert4 commented:
If you want to see the documentation on a webpage rather than just looking at text files, after you pull this branch, go into the `docs` subdirectory and do `make html`. Then the files should appear in the `_build/html` subdirectory.

bsunnquist commented on docs/configuration_files.rst:
```diff
@@ -7,92 +7,92 @@ Jwst_reffiles makes use of multiple configuration files to control the behavior
     inputfiles:
-       dark_pattern: Bias|Dark|BIAS|DARK|ZERO
-       flat_pattern: GAIN|LIN
+       dark_pattern_: Bias|Dark|BIAS|DARK|ZERO
```
should bias be here twice?

bhilbert4 replied:
Yes. This is here so that both upper and mixed-case are used when searching for filenames. This will all have to change when dealing with real JWST data, where the filenames don't have any of these strings.

bsunnquist commented on docs/configuration_files.rst:
```diff
@@ -7,92 +7,92 @@ Jwst_reffiles makes use of multiple configuration files to control the behavior
-       outrootdir: $JWST_MKREFS_OUTDIR
-       outsubdir: test1
-       runID: AUTO
+       outrootdir_: $JWST_MKREFS_OUTDIR
```
might want to specify you have to set this environment variable in the setup/installation section. If it's not set, does the code crash or does it just write to the current working directory?

bhilbert4 replied:
I'm not sure I've tried without having it defined. I'm sure it'll crash though when it tries to expand an environment variable that doesn't exist. We should make sure mkrefs.py checks that the output directory is a valid path. I've opened #46 to monitor this.

bsunnquist commented on docs/configuration_files.rst:
```diff
@@ -122,3 +122,264 @@ An example of the main configuraiton file:
+Flat Pattern
+------------
+
+List of strings that will be used to identify  flat field files within the list of input files. mkrefs will look for these strings in the filenames.
```
extra space between 'identify' and 'flat'

bhilbert4 replied:
Fixed.

bsunnquist commented on docs/configuration_files.rst:
```diff
@@ -122,3 +122,264 @@ An example of the main configuraiton file:
+Time-Obs Fits Header Keyword
+----------------------------
+
+The header keyword in the input files that contains the time of the observation. Time are needed to enforce rules when pairing darks and flats. For JWST, set this to **TIME-END**.
```
Time is*

bhilbert4 replied:
Changed to "Times are"

bsunnquist commented on docs/configuration_files.rst:
```diff
@@ -122,3 +122,264 @@ An example of the main configuraiton file:
+Optional FitsFile Header Keywords
+---------------------------------
+
+List of optional header keywords in the input files. Used for
```
guessing "Used for..." will be completed later?

bhilbert4 replied:
Added text from Armin's response.

bsunnquist commented on docs/configuration_files.rst:
```diff
@@ -122,3 +122,264 @@ An example of the main configuraiton file:
+Output Root Directory
+---------------------
+
+Path to the top level output directory for *jwst_reffiles*. Default is to define this within the **JWST_MKREFS_OUTDIR** environment variable, but any valid path is acceptable.
```
if it's not defined, does the code fail or is it just written to e.g. cwd?

bhilbert4 replied:
#46 is tracking this. A check needs to be added to mkrefs.py that the output dir is a valid path. There is error checking for the case where it is left undefined.

bsunnquist commented on docs/install.rst:
```diff
@@ -1,5 +1,36 @@
+Environment Creation
+--------------------
+
+It may be easiest to create an environment to contain *jwst_reffiles* by first creating an environment around the JWST calibration pipeline. See the installation instructions in the `JWST repo <https://github.com/spacetelescope/jwst>`_
```
yep, this strategy worked nicely for me

bsunnquist commented on docs/install.rst:
```diff
@@ -1,5 +1,36 @@
+
+1. git clone https://github.com/spacetelescope/jwst_reffiles.git
+2. cd jwst_reffiles
+3. python setup.py install
```
maybe running setup.py takes care of this, but when I used the old method (pip install .), I had to add the location of jwst_reffiles to my PYTHON_PATH in my .bashrc file for it to work. If setup.py doesn't do this, may be worth mentioning you have to do this somewhere in the install instructions.

bsunnquist commented on docs/install.rst:
```diff
@@ -1,5 +1,36 @@
+
+  mkrefs.py -vvv --cfgfile main_config_file.cfg example_readnoise_module /path/to/dark/files/*DARK*_uncal.fits /path/to/flatfield/files/NRCN815A-LIN-53650[5678]*uncal.fits
+
+To call in this way, make sure that `mkrefs.py` is in your PYTHONPATH. Otherwise, prepend `python /path/to/mkrefs/` to the call above.
```
even with the path to mkrefs in my command line call, it still didn't find it until I added it to my PYTHONPATH in my .bashrc file; but like I said above, I installed the old way listed (pip install .) so maybe this is fixed now with setup.py

bsunnquist replied:
yeah everything was executable (except mkrefs.cfg, but I don't think that matters?) and my call specified python; I'll let you know if I notice this problem again in future runs, it could always just be a mistake on my end. But yeah, after adding it to my PYTHONPATH everything ran fine.

bsunnquist commented on docs/organize_input_files.rst:
```diff
@@ -0,0 +1,24 @@
+Identification and Orgnization of Input Files
+=============================================
+
+The primary function performed by **jwst_reffiles** is to perform the organization and bookkeeping of input files to be used in the creation of reference files. The package is designed such that users can provide lists of input files (e.g. dark current files, flat field files), along with information on which types files should be used to create a given type of reference file, as well as some sim ...
```
which types of files should be used*

bhilbert4 commented:
Thanks for the comments @bsunnquist! I think I addressed all of them, and I added in some clarifying text from Armin as well. 

If you're happy, could you hit `Approve` for the review and I'll merge?
